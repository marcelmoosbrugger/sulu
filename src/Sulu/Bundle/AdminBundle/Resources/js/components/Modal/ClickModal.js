// @flow
import React from 'react';
import type {Element} from 'react';
import {action, observable} from 'mobx';
import {observer} from 'mobx-react';
import type {ModalProps} from './types';
import Modal from './Modal';

type Props = ModalProps & {
    className?: string,
    clickElement: Element<*>,
};

@observer
export default class ClickModal extends React.PureComponent<Props> {
    @observable modalOpen = false;

    @action handleElementClick = () => {
        this.modalOpen = true;
    };

    @action handleRequestClose = () => {
        this.modalOpen = false;
    };

    render() {
        const {className, clickElement, ...modalProps} = this.props;
        return (
            <div className={className}>
                {React.cloneElement(clickElement, {onClick: this.handleElementClick})}
                <Modal
                    isOpen={this.modalOpen}
                    onRequestClose={this.handleRequestClose}
                    {...modalProps}>
                    {this.props.children}
                </Modal>
            </div>
        );
    }
}
