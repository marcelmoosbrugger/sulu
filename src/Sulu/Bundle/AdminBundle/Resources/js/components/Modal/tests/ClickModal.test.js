/* eslint-disable flowtype/require-valid-file-annotation */
import React from 'react';
import {mount} from 'enzyme';
import pretty from 'pretty';
import ClickModal from '../ClickModal';

jest.mock('../../../services/DOM/afterElementsRendered');

afterEach(() => document.body.innerHTML = '');

test('The modal should initially not be rendered', () => {
    const body = document.body;
    const view = mount(
        <ClickModal
            clickElement={<button>Open modal</button>}
            title="My modal title"
            confirmText="Apply" >
            <p>My modal content</p>
        </ClickModal>
    ).render();
    expect(view).toMatchSnapshot();
    expect(body.innerHTML).toBe('');
});

test('The modal should be rendered when the button got clicked', () => {
    const body = document.body;
    const view = mount(
        <ClickModal
            clickElement={<button>Open modal</button>}
            title="My modal title"
            confirmText="Apply" >
            <p>My modal content</p>
        </ClickModal>
    );
    view.render();

    expect(body.innerHTML).toBe('');
    view.find('button').simulate('click');
    expect(pretty(body.innerHTML)).toMatchSnapshot();
});

test('The modal should be gone when the modal requests to be closed', () => {
    const view = mount(
        <ClickModal
            clickElement={<button>Open modal</button>}
            title="My modal title"
            confirmText="Apply" >
            <p>My modal content</p>
        </ClickModal>
    );
    view.find('button').simulate('click');
    view.find('Modal').props().onRequestClose();
    expect(view.find('Modal').props().isOpen).toBe(false);
});
