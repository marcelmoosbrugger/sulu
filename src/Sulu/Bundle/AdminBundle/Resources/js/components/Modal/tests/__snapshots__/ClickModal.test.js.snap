// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The modal should be rendered when the button got clicked 1`] = `
"<div>
  <div data-reactroot=\\"\\" class=\\"container isDown\\">
    <div class=\\"box\\">
      <div class=\\"box\\">
        <div class=\\"header\\">
          <!-- react-text: 5 -->My modal title
          <!-- /react-text --><span class=\\"icon fa fa-times\\" aria-hidden=\\"true\\"></span></div>
        <div class=\\"content\\">
          <p>My modal content</p>
        </div>
        <div class=\\"footer\\">
          <button class=\\"confirmButton\\">Apply</button>
        </div>
      </div>
    </div>
    <div class=\\"backdrop isVisible\\"></div>
  </div>
</div>"
`;

exports[`The modal should initially not be rendered 1`] = `
<div>
  <button>
    Open modal
  </button>
  <div>
    
  </div>
</div>
`;
